File: backend/jobtracker/src/main/java/com/clara/jobtracker/application/dto/ApplicationResponseDto.java
package com.clara.jobtracker.application.dto;

import com.clara.jobtracker.application.enums.JobType;
import com.clara.jobtracker.application.enums.Priority;
import com.clara.jobtracker.application.enums.WorkMode;
import com.clara.jobtracker.applicationStatusHistory.enums.Status;

import java.time.LocalDate;

public record ApplicationResponseDto(
        Long id,
        String position,
        String company,
        String location,
        JobType jobType,
        Priority priority,
        Status currentStatus,
        LocalDate dateApplied,
        String notes,
        String requirements,
        WorkMode workMode
) {
}

File: frontend/src/app/features/applications/models/application-response.dto.ts
import { JobType } from '../types/JobType';
import { PriorityType } from '../types/PriorityType';
import { StatusType } from '../types/StatusType';
import { WorkModeType } from '../types/WorkModeType';

export default interface ApplicationResponseDto {
  id: number;
  position: string;
  company: string;
  location: string;
  jobType: JobType;
  priority: PriorityType;
  currentStatus: StatusType;
  dateApplied: Date;
  lastActivityAt: Date;
  notes: string;
  requirements: string;
  workMode: WorkModeType;
}
